### DISCUSSION TITLE
Week 2 post

### DISCUSSION THREAD

**Discussion by Skye404** [USU-ITLS-x265-23-summer/class-repo#48] [General]
This week was very challenging, and I still have a couple more exercises to complete. 

I would say that the HTML exercise (the Cat App) was way more mentally challenging because I was able to follow it and figure it out, but only with major mental focus and effort. I did have to find help online for some of the tasks, but I feel like I have a basic, working understanding of HTML now.

The JavaScript assignments in Exercism were challenging because I didn't feel like they were written in an easy to understand way. Every task was new and foreign. JavaScript is like reading another language, and I needed an interpreter. I didn't feel like the instructions were basic enough for a newbie like me to easily know what to do to complete the task. I honestly feel like I'm jumping through hoops with JavaScript, but nothing is sticking. If I have to apply what I've learned this week to my own project, I don't know if I would be able to with JavaScript. I have taken a ton of notes, but I don't understand them in the same way I do with the HTML notes I've taken. Like derekscholes, I found these YouTube videos helpful for completing the JavaScript assignments: https://www.youtube.com/watch?v=Xgrm-5_2mEA&list=PLJO7vAdVP8x1XreVWnTEtySN9BE8-Tr4o 

I would really like to understand the basics of JavaScript still. I'm struggling to understand the WHY behind functions. I'd like to see the real time results of my code, like the Cat App exercise had with the code on one side of the screen and the web page I created on the other.

**Comment by derekscholes** [comment]
I think next weeks assignments should help solidify some of the more difficult issues with week 2. I think week three should have been first (or at least the freecodecamp link) because it would have helped a lot with the JS on exercism.

**Comment by bst3ck** [comment]
I find that seeing my results in real time very helpful as well. One of the tools that I used heavily when debugging my code was using console.log or console.dir. When running my code, the results they provided back to me helped a little, but in order to see what I was doing in my script, I could add a console.log and see the exact value at that point in the code.

This was extremely helpful when working on the recursive array problem. Simply by adding this line of code, I was able to see where my code was erroring out.
`console.log("birdsPerday value at position " + i + ": " + birdsPerDay[i]);`

If we do any additional Exercisms, I would recommend using this to help give real-time feedback on your code. Anyway, I hope that helps.

**Comment by Skye404** [comment]
Thanks, I'll try it out!
